packages/syft/src/syft/core/tensor/fixed_precision_tensor.py

```diff
@@ -3,84 +3,181 @@
 
 # stdlib
 from typing import Any
+from typing import Dict
+from typing import List
 from typing import Optional
+from typing import Union
 
 # third party
-from google.protobuf.reflection import GeneratedProtocolMessageType
 import numpy as np
 
 # relative
-from ...proto.core.tensor.fixed_precision_tensor_pb2 import (
-    FixedPrecisionTensor as FixedPrecisionTensor_PB,
-)  # type: ignore
-from ..common.serde.deserialize import _deserialize as deserialize
 from ..common.serde.serializable import serializable
-from ..common.serde.serialize import _serialize as serialize
+from .config import DEFAULT_FLOAT_NUMPY_TYPE
+from .config import DEFAULT_INT_NUMPY_TYPE
 from .passthrough import PassthroughTensor  # type: ignore
+from .passthrough import is_acceptable_simple_type  # type: ignore
+from .smpc import context
 
 
-@serializable()
+@serializable(recursive_serde=True)
 class FixedPrecisionTensor(PassthroughTensor):
+
+    __attr_allowlist__ = ("child", "_base", "_precision", "_scale")
+
     def __init__(
-        self, value: Optional[Any] = None, base: int = 10, precision: int = 3
+        self, value: Optional[Any] = None, base: int = 2, precision: int = 16
     ) -> None:
+
         self._base = base
         self._precision = precision
         self._scale = base**precision
         if value is not None:
-            fpt_value = self._scale * value
-            encoded_value = fpt_value.astype(np.int64)
-            super().__init__(encoded_value)
+            # TODO :Should modify to be compatiable with torch.
+
+            super().__init__(self.encode(value))
+        else:
+            super().__init__(None)
+
+    def encode(self, value) -> Any:
+        value = np.array(value, DEFAULT_INT_NUMPY_TYPE)
```

Comment:
Q + NITPICK: Can't we directly do:
`
encoded_value = np.array(self._scale * value, DEFAULT_INT_NUMPY_TYPE)
return encoded_value
`

Reply:
Done